pkg/controller/provider/web/base/handler.go

```diff
@@ -84,25 +90,26 @@ func (h *Handler) Link(path string, params Params) string {
 
 //
 // Set the provider.
-func (h *Handler) setProvider(ctx *gin.Context) int {
+// Set the Provider field and the X-Provider header.
+func (h *Handler) setProvider(ctx *gin.Context) (status int) {
 	var found bool
+	uid := ctx.Param(ProviderParam)
 	h.Provider = &api.Provider{
 		ObjectMeta: meta.ObjectMeta{
-			UID: types.UID(ctx.Param(ProviderParam)),
+			UID: types.UID(uid),
 		},
 	}
 	if h.Provider.UID != "" {
 		if h.Reconciler, found = h.Container.Get(h.Provider); !found {
-			return http.StatusNotFound
+			status = http.StatusNotFound
+			return
 		}
+		ctx.Header(ProviderHeader, uid)
```

Comment:
This is part of the "Better handling of provider not ready."

The web endpoint handler sets `X-Provider` so the client can distinguish between the provider not found (404) and the resource within the provider not found (404).